eng/build.sh

```diff
@@ -58,13 +58,104 @@ usage()
   echo "Arguments can also be passed in with a single hyphen."
 }
 
+initDistroRid()
+{
+    source $scriptroot/native/init-distro-rid.sh
+
+    local passedRootfsDir=""
+    local buildOs="$1"
+    local buildArch="$2"
+    local isCrossBuild="$3"
+    # For RID calculation purposes, say we are always a portable build
+    local isPortableBuild=1
+
+    # Only pass ROOTFS_DIR if __DoCrossArchBuild is specified.
+    if (( isCrossBuild == 1 )); then
+        passedRootfsDir=${ROOTFS_DIR}
+    fi
+    initDistroRidGlobal ${buildOs} ${buildArch} ${isPortableBuild} ${passedRootfsDir}
+}
+
 arguments=''
 cmakeargs=''
 extraargs=''
 build=false
 buildtests=false
 subsetCategory=''
 checkedPossibleDirectoryToBuild=false
+crossBuild=0
+os=''
+arch=''
+
+
+# Use uname to determine what the OS is.
+OSName=$(uname -s)
+case "$OSName" in
```

Comment:
Could we factor out CpuName and OSName blocks from https://github.com/dotnet/runtime/blob/bc5e7e36cefd1b53b654419cfd2bffc6b65dd403/eng/native/build-commons.sh#L240 in a separate file (something like `eng/native/initOSAndArch.sh`) and reuse? It used to be in four places and recently converged in a common script.